core/auth/fake/interfaces/identifier.go

```diff
@@ -0,0 +1,127 @@
+package interfaces
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+
+	"flamingo.me/flamingo/v3/core/auth"
+	"flamingo.me/flamingo/v3/core/auth/fake/domain"
+	"flamingo.me/flamingo/v3/framework/config"
+	"flamingo.me/flamingo/v3/framework/flamingo"
+	"flamingo.me/flamingo/v3/framework/web"
+)
+
+type (
+	// Identifier is the fake Identifier implementation
+	Identifier struct {
+		responder     *web.Responder
+		broker        string
+		reverseRouter web.ReverseRouter
+		eventRouter   flamingo.EventRouter
+	}
+
+	fakeConfig struct {
+		Broker           string                `json:"broker"`
+		LoginTemplate    string                `json:"loginTemplate"`
+		ValidatePassword bool                  `json:"validatePassword"`
+		ValidateOtp      bool                  `json:"validateOtp"`
+		UsernameFieldID  string                `json:"usernameFieldId"`
+		PasswordFieldID  string                `json:"passwordFieldId"`
+		OtpFieldID       string                `json:"otpFieldId"`
+		UserConfig       map[string]userConfig `json:"userConfig"`
+	}
+
+	userConfig struct {
+		Password string `json:"password"`
+		Otp      string `json:"otp"`
+	}
+)
+
+// FakeAuthURL - URL to fake login page
+const FakeAuthURL string = "/core/auth/fake/:broker"
+
+var (
+	_ auth.RequestIdentifier = (*Identifier)(nil)
+	_ auth.WebCallbacker     = (*Identifier)(nil)
+	_ auth.WebLogouter       = (*Identifier)(nil)
+
+	identifierConfig map[string]fakeConfig
+)
+
+func init() {
+	identifierConfig = make(map[string]fakeConfig)
+}
+
+// FakeIdentityProviderFactory -
+func FakeIdentityProviderFactory(cfg config.Map) (auth.RequestIdentifier, error) {
+	var fakeConfig fakeConfig
+
+	if err := cfg.MapInto(&fakeConfig); err != nil {
+		return nil, err
+	}
+
+	identifierConfig[fakeConfig.Broker] = fakeConfig
+
+	return &Identifier{broker: fakeConfig.Broker}, nil
+}
+
+// Inject injects module dependencies
+func (i *Identifier) Inject(
+	reverseRouter web.ReverseRouter,
+	eventRouter flamingo.EventRouter,
+) *Identifier {
+	i.reverseRouter = reverseRouter
+	i.eventRouter = eventRouter
+
+	return i
+}
+
+// Broker returns the broker id from the config
+func (i *Identifier) Broker() string {
+	return i.broker
+}
+
+// Authenticate action, fake
+func (i *Identifier) Authenticate(_ context.Context, r *web.Request) web.Result {
+	authURL, err := i.reverseRouter.Absolute(r, "core.auth.fake.auth", map[string]string{"broker": i.broker})
```

Comment:
Why does the auth url redirects to the controller, and not handle the request by itself?